clients/addoninstance/addoninstance.go

```diff
@@ -0,0 +1,33 @@
+package addoninstance
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+// SetAddonInstanceCondition sets a certain condition on the AddonInstance corresponding to the provided Addon
+// this function can be used by our tenants to report a heartbeat as well
+func SetAddonInstanceCondition(ctx context.Context, cacheBackedKubeClient client.Client, condition metav1.Condition, addonName string) error {
```

Comment:
Just a thought, if the developer already knows the `addonName` (which is internal RedHat managed-tenants knowledge), it probably means the developer should know the `targetNamespace`? Maybe we can skip parsing the namespace from the addon?

How is the `AddonInstance` namespace determined?

Reply:
The AddonInstance namespace is determined from the targetNamespace (spec.install.olmOwnNamespace/olmAllNamespaces.namespace) of the addon it corresponds to :)

You're right regarding it being possible for the Addon's developer to dig the target namespace and feed it alongside addonName but the focus was to keep things as abstract and convenient for them as possible and give them as less of a scope to make any errors which might arise, during the local development of their addon, if they feed a targetNamespace incoherent to the addonName. Also, the repetitive Reads made by the "GET addon(addonName)" call made by this function will be performed by the controller-runtime manager's client, on the Addon's side and that is backed its own informer cache, hence, the performance implications pretty negligible here :)

 Rest, I still totally see the sense in your point. I mean technically, the above function isn't even accomplishing something super big, considering it being just a simple "Update" operation on an AddonInstance resource whose name is "addon-instance" (always) and namespace is the targetNamespace of the addon, something which even the addon Devs can code from scratch.
But yeah, as I mentioned, the intentions were to keep it as abstract as possible for the Devs :)